src/core/qgsarchive.cpp

```diff
@@ -95,14 +95,27 @@ void QgsArchive::addFile( const QString &file )
   mFiles.append( file );
 }
 
+bool QgsArchive::removeFile( const QString &file )
+{
+  bool rc = false;
+
+  if ( !file.isEmpty() && QFile::exists( file ) )
+    rc = QFile::remove( file );
+
+  if ( rc )
```

Comment:
What is this check here for?
That's probably happening if a) the file is empty, b) the file doesn't exit, c) the file is write protected d) some other process blocking the rm operation or similar.

In such cases, what's the "recovery scenario"?